tests/reference/tests/chocolateysource/apply_manifest_without_location.rb

```diff
@@ -0,0 +1,24 @@
+require 'chocolatey_helper'
+test_name 'MODULES-3037 - Add Source: Apply manifest without location'
+confine(:to, :platform => 'windows')
+
+backup_config
+
+# arrange
+chocolatey_src = <<-PP
+  chocolateysource {'chocolatey':
+    ensure   => present,
+  }
+PP
+
+# teardown
+teardown do
+  reset_config
+end
+
+# act
+step 'Apply Manifest'
+apply_manifest(chocolatey_src, :catch_failures => true) do
+  step 'Verify Result'
+  assert_match(/Notice: \/Stage\[main\]\/Main\/Chocolateysource\[chocolatey\]\/location: location changed 'https:\/\/chocolatey.org\/api\/v2\/' to 'chocolatey'/, stdout, "stdout did not match expected")
```

Comment:
@ferventcoder  @Iristyle The behavior here of declaring a chcolateysource with no location and having the chocolatey.config location overwritten with the resource name seems bad. Why would you ever want that to happen? I believe it should just throw an error.

Ex: before the apply the source looks like `<source id="chocolatey" value="https://chocolatey.org/api/v2/" disabled="false" priority="0" />`

After it loos like ` <source id="chocolatey" value="chocolatey" disabled="false" priority="0" />`